File: StepsTravesal/java/StepsTraversal.java
import java.util.*;
import java.time.*;

public class StepsTraversal {
	
	public static void main(String[] args) {
		Instant start = Instant.now();
		System.out.println(jump(30));
		Instant end = Instant.now();
		System.out.println(Duration.between(start, end));
	}
	
	public static long jump(int remainingSteps) {
		if (remainingSteps > 0) {
			return jump(remainingSteps - 1) + jump(remainingSteps - 2) + jump(remainingSteps - 3);
		} else {
			return 1;
		}
	}
	
}

File: StepsTravesal/go/StepsTraversal.go
package main

import "fmt"
import "time"

func jump(steps int) int64 {
	if steps > 0 {
	return jump(steps - 1) + jump(steps - 2) + jump(steps - 3)
	} else {
		return 1
	}
}

func main() {
	start := time.Now()
	fmt.Println(jump(30))
	elapsed := time.Since(start)
    	fmt.Println("Took ", elapsed)
}
